### ISSUE TITLE
How is the confidence score used in VQA training?

### ISSUE THREAD

**Issue by chenxwh** [OFA-Sys/OFA#84]
Hi, I have a question regarding how the confidence score associated with the annotated answer is used for VQA? I don't think it is mentioned in the paper, could you explain for example, how would the confidence from `0.3!|+ans1` and `1.0!|+ans2` be processed, which part of the code is related to this? I am not entirely sure the mechanism of training with the confidence then just assigning '1.0!|+' in for inference.

Also, is it possible to generate top K answers instead of just one?

Thank you!

**Comment by yangapku** [comment]
Hi, during training, we take the confidence score as the weight of loss, which means an answer like `0.3!|+ans1` will result in a factor of 0.3 multiplied to the original cross-entropy loss for this sample. Please refer to the `get_lprobs_and_target()` method in `label_smoothed_cross_entropy.py`. During validation, the confidence score of ground truth answer is used to compute the validation score. Please refer to the official evaluation metrics of VQA challenge. For inference on test set, since  we do not have ground truth answers for testing samples, the answer field in TSV file is just a dummy placeholder.

For all-candidate evaluation, you can easily get top-K answers by replacing the `argmax` operation (across the dimension of answer candidates) with top-K selection operation in this line `predicts = valid_result.argmax(1).tolist()` (in `eval_vqa_gen()` method of `eval_utils.py`), which will return the top-K answer indices. For beam-search inference, just replace `detok_hypo_str = decode_fn(hypos[i][0]["tokens"][prefix_len:], task.tgt_dict, task.bpe, generator)` (also in `eval_vqa_gen()`) with iteration on `hypos[i][j]` will get you there.
